engine/pom.xml

```diff
@@ -218,8 +218,10 @@
                             org.hibernate.search.query;version="${project.version}",
                             org.hibernate.search.query.collector;version="${project.version}",
                             org.hibernate.search.query.dsl;version="${project.version}",
+                            org.hibernate.search.query.dsl.sort;version="${project.version}",
                             org.hibernate.search.query.engine;version="${project.version}",
                             org.hibernate.search.query.engine.spi;version="${project.version}",
+                            org.hibernate.search.query.engine.spi.impl;version="${project.version}",
```

Comment:
I understand (and agree with) what you're saying, and yes I might have got a bit carried away with exposing an `impl` package (even for an SPI implementation), but I do not quite follow what you're suggesting.

Anyway, I'll make a proposal that should rid us of exposing the LuceneQueryDescriptor altogether.